tests/common/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Copyright (C) 2019 HERE Europe B.V.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# License-Filename: LICENSE
+
+cmake_minimum_required(VERSION 3.5)
+
+add_library(olp-cpp-sdk-tests-common-lib INTERFACE)
```

Comment:
Actually, adding -lib suffix might be handy - it is easier to switch between targets in IDE - you better understand where is the executable and where is the library.

Reply:
I recall a strong desire to avoid placeholder files and now this library has nothing that could be compiled in it so I've made it a header only. 

Functional and integration tests use -lib suffix in target names and that's ok.

Reply:
This common lib will disappear anyway as the cache mock will disappear.
And even if you want to keep it, by some strange reason, you should, as recommended by gmock and also mentioned in the confluence page, put the destructor and constructor of the mock in a cpp file.

@borzun So you want to tell me that instead of using your IDE to get the build targets available, you rather add an lib at the end of an lib? libolp-cpp-sdk-tests-common-lib just to know that it is a lib in IDE? In a public repository?

Reply:
@andescu 
What I wanted to say is the pros of having such prefix. And keep in mind, that this approach with `-lib` suffix, is currently used in our repo.